raven/processes/entrypoint_regionalization.py

```diff
@@ -0,0 +1,172 @@
+)
+from regionalization_tools import (
+        get_array_of_gauged_properties,
+        filter_array_by_NSE,
+        rank_donor_catchments,
+        get_ungauged_properties,
+        IDW
+)
+
+
+def Regionalisation(hydrological_model,
+                               inputs_file,
+                               regionalization_method=2,
+                               number_donors=1,
+                               min_NSE=0.6):
+    
+    # Find the properties we wish to use for multiple linear regression (MLR)
+    # and similarity.
+    properties_to_use = ['centroid_latitude',
+                         'centroid_longitude',
+                         'surface_area',
+                         'avg_elevation',
+                         'avg_slope',
+                         'land_forest',
+                         'land_grass',
+                         'land_impervious',
+                         'land_urban']
+
+    # Get the ungauged catchment properties from the inputs_file for the
+    # regionalization scheme.
+    ungauged_properties = get_ungauged_properties(inputs_file,properties_to_use)
+
+
+    # Load gauged catchments properties and hydrological model parameters.
+    gauged_properties, model_parameters = get_array_of_gauged_properties(
+                                            hydrological_model)
+
+    # Filter arrays according to the minimum value allowed.
+    gauged_properties, model_parameters = filter_array_by_NSE(
+                                            gauged_properties,
+                                            model_parameters,
+                                            min_NSE)
+
+    # Check to see if we have enough data, otherwise raise error
+    if (gauged_properties.shape[0] < number_donors and
+            regionalization_method >= 2):
+        raise ValueError("hydrological_model and minimum NSE threshold \
+                         combination is too strict for the number of donor \
+                         basins. Please reduce the number of donor basins OR \
+                         reduce the minimum NSE threshold")
+
+    # Rank the matrix according to the similarity or distance.
+    if (regionalization_method >= 5):  # physical similarity
+        regionalization_tag = 'similarity'
+    else:
+        # Filter by distance if MLR (option 1) is selected, it is faster
+        # than by similarity.
+        regionalization_tag = 'distance'
+
+    # Create a dataframe that is identical to gauged_properties arrays.
+    gauged_properties, model_parameters = rank_donor_catchments(
+                                            ungauged_properties,
+                                            model_parameters,
+                                            gauged_properties,
+                                            properties_to_use,
+                                            regionalization_tag
+                                            )
+
+    if regionalization_method == 1:

```

Comment:
I suggest giving names to the different methods.

Reply:
OK will do that soon.